src/Analysis/Engine/Impl/Infrastructure/Extensions/StringBuilderExtensions.cs

```diff
@@ -25,19 +25,19 @@ internal static class StringBuilderExtensions {
             return sb;
         }
 
-        public static StringBuilder EnsureEndsWithSpace(this StringBuilder sb, int count = 1, bool allowLeading = false) {
```

Comment:
Current implementation it doesn't always **add**. It ensures that if there is no enough white spaces at the end, it will add them using ` `. It won't replace existing tab or non-latin space with the ` `. 

As for the name, it is `WhiteSpace` across .net standard (e.g.: `char.IsWhiteSpace`).

Reply:
Okay.